### DISCUSSION TITLE
lfric_atm mesh issues (example run)

### DISCUSSION THREAD

**Discussion by krisztiankohary** [MetOffice/simulation-systems#189] [LFRic]
Hello,
- On our local development server epic.astro (RHEL9) at the University of Exeter we may now have something that can be useful with regards to run/use LFRic
- Managed to run gungho_model app in example directory (although I have not checked science just yet) both with 1 and 6 processors
- lfric_atm app seems to be working as well (example directory) when using one(!) processor 
- But we seem to have a mesh related issue when trying to run with 16 procs
[err.txt](https://github.com/MetOffice/simulation-systems/files/15306936/err.txt)
- These are the commands to make and run the code 
[MakeAndMPIexecCommands.txt](https://github.com/MetOffice/simulation-systems/files/15306945/MakeAndMPIexecCommands.txt)
-  Any ideas how to resolve this please?

Many thanks
Krisztian

**Comment by mike-hobson** [comment]
Hi Krisztian,
Are you running with the mesh from the lfric_atm example directory? If so, I think that is a really small 2x2 planar mesh that can't be partitioned over 16 processors.

**Reply by krisztiankohary** [reply]
Hi Mike, thanks for looking into this! 

Yes, I try to run the code from the example directory. I tried to reduce the number of processors. With '1', the code seems to be running fine. However, with -np of '2' (or '4') I get a YAXT error message of

_Fatal error in /path/to/UoEGCC1120SWstackBuild/build/yaxt-0.9.2.1/src/xt_xmap_intersection_ext.c, line 448: ERROR: destination intersections do not match with destination index list (first missing index 0 at position 1)
application called MPI_Abort(comm=0xC4000014, 1) - process 0
Fatal error in /path/to/UoEGCC1120SWstackBuild/build/yaxt-0.9.2.1/src/xt_xmap_intersection_ext.c, line 448: ERROR: destination intersections do not match with destination index list (first missing index 0 at position 1)
application called MPI_Abort(comm=0xC400000A, 1) - process 1_

**Reply by mike-hobson** [reply]
Yes I'm not surprised by this. If you partition either the x- or y-direction of a 2x2 mesh with 2 processors, then the halo cells around each partition will wrap around, so the halo on each side is actually the same cell. YAXT is a library that performs our halo exchanges and it is understandably having difficulty with this odd setup. If you want to run in parallel, I suggest using a larger mesh than a 2x2.

**Reply by krisztiankohary** [reply]
Thanks Mike! Is there a 'quick' "functionality" test that I can run to test lfric_atm is working in 'parallel' mode please? I'll ask group members here about the mesh.

**Reply by mike-hobson** [reply]
I'm afraid the model is designed to be run through the Cylc suites. We include a very simple serial configuration that allows you to test you've got everything built properly, but if you want anything else, you'll have to design your own configuration or extract something from one of the suite jobs.

**Reply by krisztiankohary** [reply]
Thanks Mike, this is great, and I will look into running cylc suites on our development server at the University of Exeter! 
(I was just hoping that there may be a quick command-line functionality testing option...)
